specification/scvmm/ScVmm.Management/models.tsp

```diff
@@ -640,6 +643,36 @@ model OsProfileForVmInstance {
   /** Gets os version. */
   @visibility("read")
   osVersion?: string;
+
+  /** Gets or sets the domain name. */
+  @added(Versions.v2024_06_01)
+  domainName?: string;
+
+  /** Gets or sets the domain username. */
+  @added(Versions.v2024_06_01)
+  domainUsername?: string;
+
+  /** Password of the domain the VM has to join. */
+  @visibility("create", "update")
+  @secret
+  @added(Versions.v2024_06_01)
+  domainPassword?: string;
+
+  /** Gets or sets the workgroup. */
+  @added(Versions.v2024_06_01)
+  workgroup?: string;
+
+  /** Gets or sets the product key.Input format xxxxx-xxxxx-xxxxx-xxxxx-xxxxx */
+  @added(Versions.v2024_06_01)
+  productKey?: string;
```

Comment:
Any @pattern here?

Reply:
There is an expected format for the data received from portal however there is no such check as the command itself will throw the error of invalid product key.